main.tf

```
@@ -47,24 +60,25 @@ resource "aws_internet_gateway" "main" {
   vpc_id = aws_vpc.main.id
 
   tags = {
-    Name = "main"
+    Name = "main${aws_vpc.main.id}"
   }
 }
 
 resource "aws_eip" "nat" {
-  count = 2
+# here if I replace count, I am not sure with what value have to be in lenght().
+  count = lenght(aws_eip.nat)

```

It is ok to also connect this with `local.private_cidr`
```suggestion
  count = length(local.private_cidr)

```